tortoise_pathway/state.py

```diff
@@ -122,127 +103,146 @@ def prev(self) -> "State":
         # Remove the model if it exists
-        if model_name in self._schema["models"]:
-            del self._schema["models"][model_name]
+        if operation.model_name in app_models:
+            del app_models[operation.model_name]
 
     def _apply_rename_model(self, operation: RenameModel) -> None:
         """Apply a RenameModel operation to the state."""
-        model_name = operation.model_name
-        model = self._schema["models"][model_name]
+        app_models = self._get_app_models(operation.app_name)
+        model = app_models[operation.model_name]
 
         if operation.new_table_name:
             model["table"] = operation.new_table_name
 
         if operation.new_model_name:
-            del self._schema["models"][model_name]
-            self._schema["models"][operation.new_model_name] = model
+            del app_models[operation.model_name]
+            app_models[operation.new_model_name] = model
 
     def _apply_add_field(self, operation: AddField) -> None:
         """Apply an AddField operation to the state."""
         model_name = operation.model_name
         field_obj = operation.field_object
         field_name = operation.field_name
         # Add the field directly to the state
-        self._schema["models"][model_name]["fields"][field_name] = field_obj
+        app_models = self._get_app_models(operation.app_name)
+        app_models[model_name]["fields"][field_name] = field_obj
 
         # m2m fields are bidirectional, so we need to add the field to the referred model
         if isinstance(field_obj, ManyToManyFieldInstance):
             m2m_field = cast(ManyToManyFieldInstance, field_obj)
-            referred_model_name = m2m_field.model_name.split(".")[1]
-            self._schema["models"][referred_model_name]["fields"][m2m_field.related_name] = (
-                ManyToManyFieldInstance(
-                    model_name=f"{self.app_name}.{model_name}",
-                    through=m2m_field.through,
-                    related_name=field_name,
-                    on_delete=m2m_field.on_delete,
-                )
+            referred_model_app, referred_model_name = Operation._split_model_reference(
+                m2m_field.model_name
+            )
+            referred_app_models = self._get_app_models(referred_model_app)
+            referred_app_models[referred_model_name]["fields"][
+                m2m_field.related_name
+            ] = ManyToManyFieldInstance(
+                model_name=f"{operation.app_name}.{operation.model_name}",
+                through=m2m_field.through,
+                related_name=field_name,
+                on_delete=m2m_field.on_delete,
             )
 
     def _apply_drop_field(self, operation: DropField) -> None:
         """Apply a DropField operation to the state."""
         model_name = operation.model_name
         field_name = operation.field_name
 
+        model_fields = self._schema[operation.app_name]["models"][model_name]["fields"]
```

Comment:
nit: this is repeated multiple times. Maybe, move it to a method?

Reply:
Sure, I've taken a pass at consolidating some of the repeated behaviors, and also added some consistency to the get methods - raising `KeyError`s if those fields/models/apps are missing